pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/text/TextIndexConfigBuilder.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.segment.local.segment.index.text;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.pinot.segment.local.segment.store.TextIndexUtils;
+import org.apache.pinot.segment.spi.index.TextIndexConfig;
+import org.apache.pinot.spi.config.table.FSTType;
+import org.apache.pinot.spi.config.table.FieldConfig;
+
+
+public class TextIndexConfigBuilder extends TextIndexConfig.AbstractBuilder {
+  public TextIndexConfigBuilder(@Nullable FSTType fstType) {
+    super(fstType);
+  }
+
+  public TextIndexConfigBuilder(TextIndexConfig other) {
+    super(other);
+  }
+
+  @Override
+  public TextIndexConfig.AbstractBuilder withProperties(@Nullable Map<String, String> textIndexProperties) {
+    if (textIndexProperties != null) {
+      if (Boolean.parseBoolean(textIndexProperties.get(FieldConfig.TEXT_INDEX_NO_RAW_DATA))) {
+        _rawValueForTextIndex = textIndexProperties.get(FieldConfig.TEXT_INDEX_RAW_VALUE);
+        if (_rawValueForTextIndex == null) {
+          _rawValueForTextIndex = FieldConfig.TEXT_INDEX_DEFAULT_RAW_VALUE;
+        }
+      }
+      _enableQueryCache = Boolean.parseBoolean(textIndexProperties.get(FieldConfig.TEXT_INDEX_ENABLE_QUERY_CACHE));
+      _useANDForMultiTermQueries = Boolean.parseBoolean(
+          textIndexProperties.get(FieldConfig.TEXT_INDEX_USE_AND_FOR_MULTI_TERM_QUERIES));
+      _stopWordsInclude = TextIndexUtils.extractStopWordsInclude(textIndexProperties);
+      _stopWordsExclude = TextIndexUtils.extractStopWordsExclude(textIndexProperties);
+    }
+    return this;
+  }
+
+  @Override
+  public TextIndexConfigBuilder withStopWordsInclude(List<String> stopWordsInclude) {
```

Comment:
hehe. Yes, there is a reason: This class should return `TextIndexConfigBuilder` instead of `TextIndexConfig.AbstractBuilder` as it superclass does.

But what I'm not sure about is whether we actually need this class. IIRC I created the abstract builder when I though FST and Text index share more logic. But now there is only one subclass of `TextIndexConfig.AbstractBuilder`

Reply:
You are correct. I guess I had other attributes here in the past? I've just removed these methods.